meteor/client/ui/Settings/ShowStyle/VariantSettings.tsx

```diff
@@ -35,7 +42,101 @@ export const ShowStyleVariantsSettings = withTranslation()(
 
 			this.state = {
 				editedMappings: [],
+				uploadFileKey: Date.now(),
+			}
+		}
+		importVariants(e: React.ChangeEvent<HTMLInputElement>) {
+			const { t } = this.props
+
+			const file = e.target.files ? e.target.files[0] : null
+			if (!file) {
+				return
+			}
+
+			const reader = new FileReader()
+			reader.onload = (e2) => {
+				this.setState({
+					uploadFileKey: Date.now(),
+				})
+
+				const uploadFileContents = (e2.target as any).result
+
+				let newVariants: Array<ShowStyleVariant> = []
+				try {
+					newVariants = JSON.parse(uploadFileContents)
+					if (!_.isArray(newVariants)) {
+						throw new Error('Imported file did not contain an array')
+					}
+				} catch (error) {
+					NotificationCenter.push(
+						new Notification(
+							undefined,
+							NoticeLevel.WARNING,
+							t('Failed to import new showstyle variants: {{errorMessage}}', { errorMessage: error + '' }),
+							'VariantSettings'
+						)
+					)
+					return
+				}
+
+				_.forEach(newVariants, (entry) => {
+					MeteorCall.showstyles
+						.insertShowStyleVariantWithBlueprint(
```

Comment:
This method takes five parameters....
A good rule of thumb is:
0 parameter: Perfect!
1 parameter: Really nice.
2 parameters: Good but..
3 parameters: You're pushing it, but okay
4+ parameters: You are doing something wrong.

Looking at where the values for you parameters comes from then they all come from a `ShowStyleVariant`. Why not just parse the `ShowStyleVariant` as a parameter and save yourself a lot of typing?

Reply:
I'm changing this to parse a `ShowStyleVariant` instead.